easybuild/easyblocks/s/superlu.py

```diff
@@ -0,0 +1,141 @@
+
+@author: Xavier Besseron (University of Luxembourg)
+"""
+
+import os
+from distutils.version import LooseVersion
+
+import easybuild.tools.environment as env
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import get_shared_lib_ext
+from easybuild.tools.modules import get_software_root,get_software_version
+
+
+class EB_SuperLU(CMakeMake):
+    """
+    Support for building the SuperLU library
+    """
+
+    @staticmethod
+    def extra_options():
+        """
+        Define custom easyconfig parameters for SuperLU.
+        """
+        extra_vars = {
+            'build_shared_libs': [False, "Build shared library (instead of static library)", CUSTOM],
+        }
+        return CMakeMake.extra_options(extra_vars)
+
+    def configure_step(self):
+        """
+        Set the CMake options for SuperLU
+        """
+        self.cfg['separate_build_dir'] = True
+
+        if self.cfg['build_shared_libs']:
+            self.cfg.update('configopts', '-DBUILD_SHARED_LIBS=ON')
+        else:
+            self.cfg.update('configopts', '-DBUILD_SHARED_LIBS=OFF')
+
+        # Add -fPIC flag if necessary
+        if self.toolchain.options['pic']:
+            self.cfg.update('configopts', '-DCMAKE_POSITION_INDEPENDENT_CODE=ON')
+        else:
+            self.cfg.update('configopts', '-DCMAKE_POSITION_INDEPENDENT_CODE=OFF')
+
+        # Make sure not to build the slow BLAS library included in the package
+        self.cfg.update('configopts', '-Denable_blaslib=OFF')
+
+        # Set the BLAS library to use
+        # For this, use the BLA_VENDOR option from the FindBLAS module of CMake
+        # cf https://cmake.org/cmake/help/latest/module/FindBLAS.html
+        if get_software_root('imkl'):
+            imkl_version = get_software_version('imkl')
+            if LooseVersion(imkl_version) >= LooseVersion('10'):
+                self.cfg.update('configopts', '-DBLA_VENDOR="Intel10_64lp"')
+            else:
+                self.cfg.update('configopts', '-DBLA_VENDOR="Intel"')
+        elif get_software_root('ACML'):
+            self.cfg.update('configopts', '-DBLA_VENDOR="ACML"')
+        elif get_software_root('ATLAS'):
+            self.cfg.update('configopts', '-DBLA_VENDOR="ATLAS"')
+        elif get_software_root('OpenBLAS'):
+            # Unfortunately, OpenBLAS is not recognized by FindBLAS from CMake,
+            # we have to specify the OpenBLAS library manually
+            self.cfg.update('configopts', '-DBLAS_LIBRARIES="${EBROOTOPENBLAS}/lib/libopenblas.a;-pthread"')
+        else:
+            # Fallback: try everything, pick the first one
+            # CMake should fail if none is found
+            self.cfg.update('configopts', '-DBLA_VENDOR="All"')
```

Comment:
I prefer failing here.

The EB policy is to use the BLAS available in the toolchain. If it's not available, we shouldn't let SuperLU pick another random BLAS library it may find...